docs/architecture/adr-023-protobuf-naming.md

```diff
@@ -99,13 +99,67 @@ the best choices for the future.
 
 ### Versioning
 
-#### Don't Allow Breaking Changes in Stable Packages
-
-Always use a breaking change detector such as [Buf](https://buf.build) to prevent
-breaking changes in stable (non-alpha or beta) packages. Breaking changes can
-break smart contracts/persistent scripts and generally provide a bad UX for
-clients. With protobuf, there should usually be ways to extend existing
-functionality instead of just breaking it.
+#### Guidelines on Stable Package Versions
+
+In general, schema evolution is the way to update protobuf schemas. That means that new fields,
+messages, and RPC methods are _added_ to existing schemas and old fields, messages and RPC methods
+are maintained as long as possible.
+
+Breaking things is often unacceptable in a blockchain scenario. For instance, immutable smart contracts
+may depend on certain data schemas on the host chain. If the host chain breaks those schemas, the smart
+contract may be irreparably broken. Even when things can be fixed (for instance in client software),
+this often comes at a high cost.
+
+Instead of breaking things, we should make every effort to evolve schemas rather than just breaking them.
+[Buf](https://buf.build) breaking change detection should be used on all stable (non-alpha or beta) packages
+to prevent such breakage.
+
+With that in mind, different stable versions (i.e. `v1` or `v2`) of a package should more or less be considered
+different packages and this should be last resort approach for upgrading protobuf schemas. Scenarios where creating
+a `v2` may make sense are:
+* we want to create a new module with similar functionality to an existing module and adding `v2` is the most natural
+way to do this. In that case, there are really just two different, but similar modules with different APIs.
+* we want to add a new revamped API for an existing module and it's just too cumbersome to add it to the existing package,
+so putting it in `v2` is cleaner for users. In this case, care should be made to not deprecate support for
+`v1` if it is actively used in immutable smart contracts.
+
+An alternative to creating `v2` packages is creating `v2` messages and RPC methods. For instance say we found a better API
+for sending coins, then we could add `MsgSendV2` to the existing bank package without needing to change `MsgSend`
```

Comment:
I'm not sure I like adding `V2` suffix to types. We might end up with:
1. `bank.v1`'s `MsgSend` type
2. `bank.v1`'s `MsgSendV2` type
3. `bank.v2`'s `MsgSend` type
4. `bank.v2`'s `MsgSendV2` type

which could be confusing (esp 2&3). I'm okay to add Alpha/Beta suffixes to types though.

For new stable types, maybe instead of adding `V2`, we could just come up with a more descriptive name?

Reply:
Good point. Let's remove this part. Let's just keep version suffixes in the package. Modules can always support multiple versions

Reply:
agree with this point.